### PR TITLE
FEAT skeleton component ☠️

### PR DESCRIPTION
**Pull request by MarcFairbrother**
## Description

Adds a skeleton component.

### Basic usage

- the `.skeleton` class indicates that the element contains items to be displayed with a loading skeleton state
- the `.is-loading` modifier activates the loading state on all elements with a `.skeleton-item` class within `.skeleton`
- `.skeleton-item` represents body 1 text, if used on `<h*>` elements or elements with `.u-h*` or `.u-text*` size modifiers the skeleton's styles will adapt to the corresponding text size and line height

### Modifiers

- `.mod-dark` can be added to `.skeleton` or `.skeleton-item` for use on grey backgrounds
- you can use `.mod-square` or `.mod-circle` to create shapes
- `.mod-XS` to `.mod-XXL` size modifiers can be applied to the shapes
- you can control the size of the shape by using `--component-skeleton-shape-height` and `--component-skeleton-shape-width`
- you can control the width of the text skeletons by using `--component-skeleton-text-width` by setting a percentage as the value or a function that returns a random percentage

-----

### Texts and Titles

chrome_J91zUSCcLF

### Shapes

chrome_pdv9Twxfxv

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2322/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7499/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2322/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7517/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2322/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7518/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2322/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7521/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2322/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7525/ui-diff)

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7531/ui-diff)

**Review by AlexiaAlarza — APPROVED** [review]

**Comment by AlexiaAlarza** [comment]
trop hâte de le mettre en place, merci Marc !

**Comment by MarcFairbrother** [comment]
Cool @AlexiaAlarza ! Hésite pas à me faire signe quand tu commences à l'utiliser si tu rencontres des problèmes. Ca fonctionne bien dans notre utilisation chez Core mais il y aura sans doute des cas d'usage dans d'autres BU qui ne seront pas couverts. L'objectif c'est qu'on puisse faire évoluer le composant en fonction des besoins.

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2322/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7599/ui-diff)

**Review by jeremie-lucca — APPROVED** [review]

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7600/ui-diff)

File: packages/scss/src/components/skeleton/states.scss [hunk 1]
```diff
@@ -0,0 +1,85 @@
+@mixin skeletonTextSizing($sizes...) {
+	@each $size in $sizes {
+		&.u-text#{$size} {
+			@if ($size == 'XS') {
+				--components-skeleton-text-offset: calc(0.25rem);
```
Comment (BertrandPodevin) [inline packages/scss/src/components/skeleton/states.scss:5]: `calc` doesn't seems to be needed here

File: packages/scss/src/components/skeleton/states.scss [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				--components-skeleton-text-offset: 0.5rem;
+			}
+			height: var(--sizes-#{$size}-lineHeight);
+		}
+	}
+}
+
+@mixin skeletonTitleSizing($titles...) {
+	@each $title in $titles {
+		&:is(.u-#{$title}),
```
Comment (BertrandPodevin) [inline packages/scss/src/components/skeleton/states.scss:20]: `&:is(.u-#{$title}, #{$title})` ?

File: packages/scss/src/components/skeleton/states.scss [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		);
+		background-size: 150% 100%;
+		background-repeat: repeat-x;
+		display: block;
+		&:not(.mod-square, .mod-circle) {
+			@include skeletonTextSizing('XS', 'S', 'L', 'XL', 'XXL');
+			@include skeletonTitleSizing('h1', 'h2', 'h3', 'h4', 'h5', 'h6');
+			clip-path: inset(
+				calc(var(--components-skeleton-text-offset) + 1px) calc(100% - var(--components-skeleton-text-width, 100%))
+					calc(var(--components-skeleton-text-offset) - 1px) 0 round 8px
```
Comment (BertrandPodevin) [inline packages/scss/src/components/skeleton/states.scss:56]: 8px -> `--commons-borderRadius-L` ?

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2322/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7607/ui-diff)

**Review by BertrandPodevin — APPROVED** [review]

**Pull request merged**